config/config.go

```diff
@@ -314,10 +314,7 @@ func (c *Config) extend(extensionConfig Config) {
 			}
 			// The keywords from the base rule and the extended rule must be merged into the global keywords list
 			for _, k := range baseRule.Keywords {
-				c.Keywords[k] = struct{}{}
-			}
-			for _, k := range currentRule.Keywords {
```

Comment:
The `baseRule.Keywords` includes both the `baseRule.Keywords` and `currentRule.Keywords` [1](https://github.com/gitleaks/gitleaks/pull/1633/files#diff-fe44f09c4d5977b5f5eaea29170b6a0748819c9d02271746a20d81a5f3efca17R311), so when assigning the rule keywords to `config.Keywords`, we only need to iterate over the base keywords.

That means these two `for` loops essentially do the same thing, so I've removed the second loop over `currentRule.Keywords`.